src/server/helpers/collectionRouteUtils.js

```diff
@@ -34,7 +44,7 @@ export async function collectionCreateOrEditHandler(req, res, next) {
 			method = 'insert';
 		}
 		else {
-			if (upperFirst(camelCase(req.body.entityType)) !== res.locals.collection.entityType) {
+			if (res.locals.collection.items.length && (upperFirst(camelCase(req.body.entityType)) !== res.locals.collection.entityType)) {
```

Comment:
I wonder if it would be a good idea to play it a bit safer:
```suggestion
			if (res.locals.collection?.items?.length && (upperFirst(camelCase(req.body.entityType)) !== res.locals.collection.entityType)) {
```